changes/test_tasks.py

```diff
@@ -168,3 +159,52 @@ def test_non_hsm_failure(self):
         ])
 
         self.assertEqual(Change.objects.count(), 0)
+
+
+class ProcessWhatsAppSystemEventTaskTests(WhatsAppBaseTestCase):
+
+    @mock.patch('changes.tasks.utils.ms_client.create_outbound')
+    @responses.activate
+    def test_message_sent_delivered(self, mock_create_outbound):
+        """
+        The task should send the correct outbound based on the delivered event.
+        """
+        self.create_outbound_lookup()
+        self.create_identity_lookup()
+
+        process_whatsapp_system_event('messageid', "undelivered")
+
+        mock_create_outbound.assert_called_once_with({
+            'to_identity': 'test-identity-uuid',
+            'content':
+                "We see that your MomConnect WhatsApp messages are not being "
+                "delivered. If you would like to receive your messages over "
+                "SMS, reply ‘SMS’.",
+            'channel': "JUNE_TEXT",
+            'metadata': {},
+        })
+
+    @mock.patch('changes.tasks.utils.ms_client.create_outbound')
+    @responses.activate
+    def test_no_message_sent(self, mock_create_outbound):
+        """
+        The task should not create a Outbound when the event is the wrong type.
+        """
+        self.create_outbound_lookup()
+
+        process_whatsapp_system_event('messageid', "something_else")
+
+        self.assertFalse(mock_create_outbound.called)
+
+    @mock.patch('changes.tasks.utils.ms_client.create_outbound')
+    @responses.activate
+    def test_no_message_found(self, mock_create_outbound):
+        """
+        The task should not create a outbound if the original message was not
+        found
+        """
+        self.create_outbound_lookup(0)
+
+        process_whatsapp_system_event('messageid', "something_else")
```

Comment:
Shouldn't this be an `undelivered` type, so that we're actually moving on to the message processing, rather than exiting at the event type handling?

We should probably add a check to this test (maybe ensure that the outbound request was actually made) to ensure that we're actually getting that far in the processing.